src/components/game/components/board/Board.js

```diff
@@ -1,198 +1,70 @@
+import { connect } from "react-redux";
 
 import handCheck from "lib/handCheck";
-import deckCreator from "lib/deckCreator";
-
-const Board = (props) => {
-  const initialState = {
-    deck: [],
-    playerHand: [],
-    aiHand: [],
-    playerMoney: 1000,
-    aiMoney: 1000,
-    playerBet: 0,
-    aiBet: 0,
-    phase: 0,
-  };
-
-  const [state, setState] = React.useState(initialState);
-
+import { check, fold, raise, replace, startGame } from "models/game/actions";
+import {
+  aiBet,
+  aiHand,
+  aiMoney,
+  deck,
+  phase,
+  playerBet,
+  playerHand,
+  playerMoney,
+} from "models/game/selectors";
+
+const Board = ({
+  aiBet,
+  aiHand,
+  aiMoney,
+  deck,
+  phase,
+  playerBet,
+  playerHand,
+  playerMoney,
+  onClickPlayHandler,
+  onClickReplaceHandler,
+  onClickFoldHandler,
+  onClickCheckHandler,
+  onClickRaiseHandler,
+}) => {
   const renderBuildDeckButton = () => {
-    const onClickHandler = () => {
-      const newDeck = state.deck.slice();
-      const newPlayerHand = state.playerHand.slice();
-      const newAiHand = state.aiHand.slice();
-
-      const newPlayerBet = state.playerBet;
-      const newAiBet = state.aiBet;
-      const bet = 100;
-
-      const newPhase = state.phase;
-
-      newDeck.push(deckCreator());
-      const newDeckFlat = newDeck.flat();
-
-      const newerPlayerHand = newPlayerHand.concat(newDeckFlat);
-      const newerAiHand = newAiHand.concat(newDeckFlat);
-
-      if (newPhase === 0) {
-        newerPlayerHand.splice(0, 1);
-        newerPlayerHand.splice(5);
-        newerAiHand.splice(0, 6);
-        newerAiHand.splice(5);
-        newDeckFlat.splice(0, 10);
-
-        setState((prevState) => ({
-          ...prevState,
-          deck: newDeckFlat,
-          playerHand: newerPlayerHand,
-          aiHand: newerAiHand,
-          playerBet: newPlayerBet + bet,
-          aiBet: newAiBet + bet,
-          phase: newPhase + 2,
-        }));
-      } else {
-        setState((prevState) => ({ ...prevState, phase: newPhase + 1 }));
-      }
-    };
-
-    return state.phase >= 4 ? null : <Deck onClick={onClickHandler} />;
+    return phase >= 4 ? null : <Deck onClick={onClickPlayHandler} />;
   };
 
-  const renderReplaceCardButton = (handIndex) => {
-    const onClickHandler = () => {
-      const newPlayerHand = state.playerHand.slice();
-      const newDeck = state.deck.slice();
-      const newPhase = state.phase;
-
-      newPlayerHand.splice(handIndex, 1, newDeck[0]);
-      newDeck.splice(0, 1);
-
-      if (newDeck.length > 39) {
-        setState((prevState) => ({
-          ...prevState,
-          deck: newDeck,
-          playerHand: newPlayerHand,
-          phase: newPhase + 0.5,
-        }));
-      }
-    };
-
-    if (state.phase <= 2 || state.phase >= 4) {
-      return;
-    }
-
-    return <Replace onClick={onClickHandler} />;
+  const renderReplaceCardButton = () => {
+    if (phase <= 2 || phase >= 4) {
```

Comment:
This comment applies for more occurrences.
Rendering of an element should occur by the decision of the container element. For example the parent component should have the if that will allow the render of a component.

- By doing this we achieve purity (avoid using a prop that's not a part of our scope (phase in this case).
- It is reusable (you can make button component that will have props and render differently based on its props

Reply:
Done.

Reply:
this function remains impure because it checks phase var, but  phase it's not in it's scope or concern. Parent component which has access to phase should decide the button's fate and not itself.